docs/data/common-concepts/custom-components/custom-components.md

```diff
@@ -0,0 +1,244 @@
+# Custom slots and subcomponents
+
+<p class="description">Learn how to override the default DOM structure of the MUI X components.</p>
+
+## What is a slot?
+
+A slot is a part of a component that can be overridden and / or customized.
+
+## Basic usage
+
+### How to override a slot?
+
+You can override a slot by providing a custom component to the `slots` prop:
+
+{{"demo": "CustomSlot.js"}}
+
+### How to customize a slot?
+
+You can pass props to any slot using the `slotProps` prop:
+
+{{"demo": "CustomSlotProps.js"}}
+
+You can also use both `slots` and `slotProps` on the same component:
+
+{{"demo": "CustomSlotAndSlotProps.js"}}
+
+Most components also support a callback version of `slotProps`.
+This callback receives an object that contains information about the current state of the component,
+that information can vary depending on the slot being used:
+
+{{"demo": "CustomSlotPropsCallback.js"}}
+
+## Correct usage
```

Comment:
Well done, this section is very much needed! 👍🏻

Reply:
Kudos to @LukasTy , it's a pure copy paste :laughing: